### ISSUE TITLE
don't authenticate a secure route during a login

### ISSUE THREAD

**Issue by pherrmann** [salte-auth/salte-auth#140]
### Description
This a followup to [#135](https://github.com/salte-io/salte-auth/issues/135). This is a very similar issue. 

I'm getting a state mismatch error and an additional authorize call to get the access token during a loginWithRedirect call. 

This is due to having all routes secure and $onRouteChanged being triggered on the initial route

### Expected outcome

loginWithRedirect  returns no state mismatch error and makes no extra authorize calls during the login process 

### Actual outcome

loginWithRedirect returns a state mismatch error during the login process and makes an extra authorize call 

### Proposal

Don't authenticate a secure route during a login. I will open a PR with a proposed solution to resolve this issue.

**Comment by cecilia-sanare** [comment]
Hmm, I'm not exactly sure how this is happening since based 
upon the code it should be executing in the following order:

1. `SalteAuth` is initialized
2. You call `loginWithRedirect` which attaches the login promise.
3. `$$onRouteChanged` is invoked after the first event loop. This will try to automatically call login, but it will return the promise from step 2 instead.
4. After you login `retrieveAccessToken` continues attempts to retrieve an `accessToken` if it needs to.

**Comment by pherrmann** [comment]
It doesn't appear I have rights to push my fix branch to the remote

**Comment by cecilia-sanare** [comment]
In order to contribute to a repository you aren't a collaboration on you need to:

* fork it
* push your code to your version of the repo
* open a PR to the main repository.

**Comment by cecilia-sanare** [comment]
**Here's the breakdown of whats going on:**

* `salte-auth` Initializes
* Developer initiates `loginWithRedirect`
* `salte-auth` calls `retrieveAccessToken`
* Promise returned by `loginWithRedirect` resolves
* The browser finally navigates to the `/authorize` page

**I think the most reliable course of action is to change:**

https://github.com/salte-io/salte-auth/blob/885941b9395ebfbd3e6d3b6e17e839c05c85b64f/src/salte-auth.js#L303-L306

**To be:**

```js
  loginWithRedirect() {
    // ...

    // NOTE: This prevents the other login types from racing "loginWithRedirect".
    // Without this someone could potentially call login somewhere else before
    // the app has a change to redirect. Which could result in an invalid state.
    this.$promises.login = new Promise(() => {});

    // ...
  }
```

That way the `loginWithRedirect` promise never resolves.

**Comment by cecilia-sanare** [comment]
Hey, I just released the above change in [v2.1.4](https://github.com/salte-io/salte-auth/releases/tag/v2.1.4). 

Let me know how it affects your issue.

**Comment by pherrmann** [comment]
Ahh ok, I'll keep that in mind in the future if I want to open a PR.

I won't need to because your change does fix the issue! I ran the change locally and confirmed there's a single authorize call and no state mismatch error

Thanks!

**Comment by cecilia-sanare** [comment]
Awesome, good to hear! 

Closing this issue as it’s been resolved by #141.
